pkg/volume/awsebs/aws_util.go

```diff
@@ -125,6 +128,39 @@ func (util *AWSDiskUtil) CreateVolume(c *awsElasticBlockStoreProvisioner, node *
 	return name, volumeOptions.CapacityGB, labels, fstype, nil
 }
 
+// getCandidateZones finds possible zones that a volume can be created in
+func getCandidateZones(cloud *aws.Cloud, selectedNode *v1.Node, pvcName string) (sets.String, error) {
+	var err error
+	zones := sets.NewString()
+	if selectedNode != nil {
+		// For topology aware volume provisioning, node is already selected so we use the zone from
+		// selected node directly
+		if zone, ok := selectedNode.Labels[v1.LabelZoneFailureDomain]; ok {
+			klog.V(4).Infof("Using zone %q of selected node %q for PVC %q", zone, selectedNode.Name, pvcName)
+			zones.Insert(zone)
+		} else {
+			// node controller should set zone label but somehow we cannot find it, fall back to retrieve information
+			// from cloud provider
+			klog.Warningf("Selected node %q does not have zone info for PVC %q, getting zone info from provider directly", selectedNode.Name, pvcName)
+			zone, err := cloud.GetZoneByProviderID(context.TODO(), selectedNode.Spec.ProviderID)
+			if err != nil {
+				return zones, err
+			}
+			klog.V(4).Infof("Using zone %q of selected node %q for PVC %q", zone.FailureDomain, selectedNode.Name, pvcName)
+			zones.Insert(zone.FailureDomain)
```

Comment:
This is redundant. The underlying code will not use value returned from this function if `selectedNode` is set in `populateVolumeOptions` functions below. We can just return empty set from this function if `selectedNode != nil`

Reply:
I see your point, makes sense to me, and we can assume node controller will maintain zone labels here